Common/Data/Consolidators/VolumeRenkoConsolidator.cs

```diff
@@ -0,0 +1,289 @@
+                    HighRate = rate;
+                    LowRate = rate;
+                    CloseRate = rate;
+                }
+                else
+                {
+                    CloseOn = data.Time;
+
+                    if (rate > HighRate)
+                    {
+                        HighRate = rate;
+                    }
+
+                    if (rate < LowRate)
+                    {
+                        LowRate = rate;
+                    }
+
+                    CloseRate = rate;
+                }
+            }
+            else if (data.GetType() == typeof(TradeBar))
+            {
+                volume += ((TradeBar)data).Volume;
+                var open = ((TradeBar)data).Open;
+                var high = ((TradeBar)data).High;
+                var low = ((TradeBar)data).Low;
+                var close = ((TradeBar)data).Close;
+
+                if (_firstTick)
+                {
+                    _firstTick = false;
+
+                    OpenOn = data.Time;
+                    CloseOn = data.EndTime;
+                    OpenRate = open;
+                    HighRate = high;
+                    LowRate = low;
+                    CloseRate = close;
+                }
+                else
+                {
+                    CloseOn = data.EndTime;
+
+                    if (high > HighRate)
+                    {
+                        HighRate = high;
+                    }
+
+                    if (low < LowRate)
+                    {
+                        LowRate = low;
+                    }
+
+                    CloseRate = close;
+                }
+            }
+
+            _volumeLeftOver = Next(data, volume);
+        }
+
+        /// <summary>
+        /// Scans this consolidator to see if it should emit a bar due to time passing
+        /// </summary>
+        /// <param name="currentLocalTime">The current time in the local time zone (same as <see cref="BaseData.Time"/>)</param>
+        public void Scan(DateTime currentLocalTime)
+        {
+        }
+
+        /// <summary>Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.</summary>
+        /// <filterpriority>2</filterpriority>
+        public void Dispose()
+        {
+            DataConsolidated = null;
+            _dataConsolidatedHandler = null;
+        }
+
+        /// <summary>
+        /// Event invocator for the DataConsolidated event. This should be invoked
+        /// by derived classes when they have consolidated a new piece of data.
+        /// </summary>
+        /// <param name="consolidated">The newly consolidated data</param>
+        protected void OnDataConsolidated(VolumeRenkoBar consolidated)
+        {
+            DataConsolidated?.Invoke(this, consolidated);
+            _currentBar = consolidated;
+            _dataConsolidatedHandler?.Invoke(this, consolidated);
+            Consolidated = consolidated;
```

Comment:
Seems should inherit directly from `DataConsolidator` to avoid a few of these issues/duplications?

Reply:
but didn't `_currentBar` and `Consolidated` need to be set? https://github.com/QuantConnect/Lean/blob/e13066273ca9ceb8c18c1f6dc49a6567913a1dc8/Common/Data/Consolidators/RenkoConsolidator.cs#L245-L251